plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,645 @@
+    }
+
+    @Override
+    public ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        return metadata.getTables(
+                IGNITE_CATALOG,
+                IGNITE_SCHEMA,
+                tableName.orElse(null),
+                new String[] {"TABLE", "VIEW"});
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+
+        switch (typeHandle.getJdbcType()) {
+            case Types.CHAR:
+                return Optional.of(charColumnMapping(typeHandle.getColumnSize()));
+            case Types.VARCHAR:
+                return Optional.of(varcharColumnMapping(typeHandle.getColumnSize()));
+            case Types.DATE:
+                return Optional.of(ColumnMapping.longMapping(
+                        DATE,
+                        dateReadFunction(),
+                        dateWriteFunction()));
+            case Types.BOOLEAN:
+                return Optional.of(booleanColumnMapping());
+
+            case Types.TINYINT:
+                return Optional.of(tinyintColumnMapping());
+
+            case Types.SMALLINT:
+                return Optional.of(smallintColumnMapping());
+
+            case Types.INTEGER:
+                return Optional.of(integerColumnMapping());
+
+            case Types.BIGINT:
+                return Optional.of(bigintColumnMapping());
+
+            case Types.FLOAT:
+                return Optional.of(realColumnMapping());
+
+            case Types.DOUBLE:
+                return Optional.of(doubleColumnMapping());
+
+            case Types.DECIMAL:
+                int decimalDigits = typeHandle.getDecimalDigits().orElseThrow(() -> new IllegalStateException("decimal digits not present"));
+                int precision = typeHandle.getRequiredColumnSize();
+                if (getDecimalRounding(session) == ALLOW_OVERFLOW && precision > Decimals.MAX_PRECISION) {
+                    int scale = min(decimalDigits, getDecimalDefaultScale(session));
+                    return Optional.of(decimalColumnMapping(createDecimalType(Decimals.MAX_PRECISION, scale), getDecimalRoundingMode(session)));
+                }
+                precision = precision + max(-decimalDigits, 0); // Map decimal(p, -s) (negative scale) to decimal(p+s, 0).
+                if (precision > Decimals.MAX_PRECISION) {
+                    break;
+                }
+                return Optional.of(decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
+
+            case Types.BINARY:
+                return Optional.of(varbinaryColumnMapping());
+
+            case Types.OTHER:
```

Comment:
Seems like a very relaxed check that will catch "other" types too. Can we do something similar to https://github.com/trinodb/trino/blob/825e6d82cfd3dee98038f14b38a55e0d1798ae35/plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java#L436-L437 instead?

Reply:
The uuid type name in ignite is 'OTHER' too, not the 'uuid'

Reply:
`typeHandle.getJdbcType()` returns OTHER for uuid but I'm suggesting to use `typeHandle.getJdbcTypeName()` instead which returns `uuid`?

Reply:
Added a brach using `jdbcTypeName` to build columnMapping in `IgniteJdbcClient#toColumnMapping`, please help to review whether i understand correctly.

Reply:
Wow - I'm surprised that UUID get's reported as `jdbcTypeName = "other"`.
Please add a comment why it's `case "other"` instead of `case "uuid"` as people would normally expect.

This can fail in surprising ways if there are other unsupported data types which return `other`.

cc: @findepi @wendigo

Reply:
unfortunately the type like `GEOMETRY` in Ignite also type is `other`.

Reply:
Can we leave UUID out of the initial version then? Let's tackle it separately (unless we already know a solution).

Reply:
sure, i will remove this part.